avocado/core/loader.py

```diff
@@ -496,8 +496,8 @@ class FileLoader(TestLoader):
     """
 
     name = 'file'
-    __not_test_str = ("Does not look like an INSTRUMENTED test, nor is it "
-                      "executable")
+    __not_test_str = ("Not an INSTRUMENTED (avocado.Test based) "
+                      "or SIMPLE (executable) test")
```

Comment:
Well do you think it's clearer to say it's not INSTRUMENTED or (nor?) SIMPLE test? To me the original message is easier to understand as it says what it is and not what type it is not.

Also this commit reminds me I should probably update this message in the python unittest loader PR, because then this becomes "Not an INSTRUMENTED, SIMPLE, or PyUNITTEST test".

Reply:
Of course I think it's clearer, if not, I wouldn't be proposing it with such a description ;).

Anyway, this came up while I was reviewing the current problems with the loader implementation, and your comment about `PyUNITTEST` is just another confirmation of that.  In summary, a argue that `NOT_A_TEST` and others should *not* be a test type.  By turning `NOT_A_TEST` and others into "test reference resolution results", no additional updating would be necessary when new test types are added.  The relationship would change from (`=>` is "knows about"):

```
NOT_A_TEST => INSTRUMENTED, SIMPLE, PyUNITTEST
```

To

```
FileLoader.INSTRUMENTED => NOT_A_TEST
FileLoader.SIMPLE => NOT_A_TEST
AnyOtherLoader.TEST_TYPE => NOT_A_TEST
```

Anyway, I think a lot needs to be changed in the loader, but please evaluate this change given the current state of things.

Reply:
Well I'd prefer one failure per loader, ideally describing why is it not that loader test along with description. The original description suited well as it described why it was not identified as a test. This description only tells it's not INSTRUMENTED nor SIMPLE test, which doesn't explain much.

In verbose mode I'd prefer something like:

```
!FILELOADER: foo.py: Avocado nor unittest import were not found and the file is not executable
!FILELOADER: bar.sh: Failed to parse as python and the file is not executable
!FILELOADER: symlink: Broken symlink
```
But that is a bit too verbose, so `Does not look like INSTRUMENTED, nor is it executable` seems informative enough. Anyway I'm fine with both (as far as you mention the `executable`) but I'd prefer to keep the `does not look like` as it could just be a broken instrumented test and it might confuse people to tell this is not an instrumented test.

Reply:
@ldoktor I'm preparing a document describing the problems we have with the loader implementation.  For instance, `!FILELOADER` is **wrong** because it uses a test type for something that is **not** a test type.

Now, this is a related, but **different** issue.  I still don't get why you insist on not having clear(er) messages.  Examples:

1) Avocado doesn't support "executables" as a test type, it supports SIMPLE tests (which implementation is satisfied by an executable
2) In a situation where "it could just be a broken instrumented test", it really means that it's **not** a INSTRUMENTED test.  It will be, if it's fixed and made into one.  There's no middle grounds.

Thing of internet name resolution.   You don't send a DNS query, and get a "maybe" RR (resource record).  We should really attempt to be clearer here.

Reply:
It's not a type, it's a message. If you remember I worked on this area and my resolution was that normal verbosity reports testsuite, increased verbosity reports records about possible (not)discovered things. Alternative is to have `discover` and `search` which would report such things. Another alternative is to report failures differently, but then you lose the order of the messages vs. discovered tests. Anyway looking forward to your RFC, perhaps it'll clear some things.

As for the message, I said I'm fine with this. The explicit `executable` is important to me, because beginners might not know what SIMPLE test means.

Anyway I already mentioned I'm fine with this, you prefer it so please update the message to reflect PyUNITTEST test type (which has been merged already) and we can get it in.